x/ref/lib/vdl/codegen/json/const.go

```diff
@@ -0,0 +1,116 @@
+// Package json implements JSON generation for VDL const values.
+package json
+
+import (
+	"fmt"
+	"strconv"
+
+	"v.io/v23/vdl"
+	"v.io/x/ref/lib/vdl/codegen"
+	"v.io/x/ref/lib/vdl/vdlutil"
+	"v.io/x/ref/lib/vdl/codegen/vdlgen"
+)
+
+// Const returns a JSON representation of a value.
+//
+// TODO(razvanm): use this function to replace most (all?) of the untypedConst
+// function from v.io/x/ref/lib/vdl/codegen/javascript.
+func Const(v *vdl.Value, pkgPath string, imports codegen.Imports) string {
+	switch v.Kind() {
+	case vdl.Bool:
+		if v.Bool() {
+			return "true"
+		}
+		return "false"
+	case vdl.Byte, vdl.Uint16, vdl.Uint32:
+		return strconv.FormatUint(v.Uint(), 10)
+	case vdl.Int8, vdl.Int16, vdl.Int32:
+		return strconv.FormatInt(v.Int(), 10)
+	case vdl.Uint64:
+		return fmt.Sprintf(`"%d"`, v.Uint())
+	case vdl.Int64:
+		return fmt.Sprintf(`"%d"`, v.Int())
+	case vdl.Float32, vdl.Float64:
+		return strconv.FormatFloat(v.Float(), 'g', -1, bitlen(v.Kind()))
+	case vdl.String:
+		return strconv.Quote(v.RawString())
+	case vdl.Any:
+		if elem := v.Elem(); elem != nil {
+			return Const(elem, pkgPath, imports)
+		}
+		return "null"
+	case vdl.Optional:
+		if elem := v.Elem(); elem != nil {
+			return Const(elem, pkgPath, imports)
+		}
+		return "null"
+	case vdl.Enum:
+		return fmt.Sprintf(`"%s"`, v.EnumLabel())
```

Comment:
Optionally, do this instead:
```
  return strconv.Quote(v.EnumLabel())
```

I honestly don't remember why the javascript package uses double-quotes for vdl.String, but single-quotes for vdl.Enum.  Enum labels shouldn't have unprintable characters (so that's not a concern), but nonetheless having consistency between vdl.String and vdl.Enum handling seems nicer.

Unless, I guess, if enum labels might have embedded double-quotes; maybe that's why we used single-quotes in the `javascript` package?